src/pages/home/report/LinkPreviewer.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import {View, Image} from 'react-native';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import {uniqBy} from 'lodash';
+import Text from '../../../components/Text';
+import styles from '../../../styles/styles';
+import variables from '../../../styles/variables';
+
+const IMAGE_TYPES = ['jpg', 'jpeg', 'png'];
+
+const propTypes = {
+    /** Data about links provided in message. */
+    data: PropTypes.arrayOf(
+        PropTypes.shape({
+            /**  The URL of the link. */
+            url: PropTypes.string,
+
+            /**  A description of the link. */
+            description: PropTypes.string,
+
+            /**  The title of the link. */
+            title: PropTypes.string,
+
+            /**  The publisher of the link. */
+            publisher: PropTypes.string,
+
+            /**  The image associated with the link. */
+            image: PropTypes.shape({
+                /**  The height of the image. */
+                height: PropTypes.number,
+
+                /**  The width of the image. */
+                width: PropTypes.number,
+
+                /**  The URL of the image. */
+                url: PropTypes.string,
+            }),
+
+            /**  The provider logo associated with the link. */
+            logo: PropTypes.shape({
+                /**  The height of the logo. */
+                height: PropTypes.number,
+
+                /**  The width of the logo. */
+                width: PropTypes.number,
+
+                /**  The URL of the logo. */
+                url: PropTypes.string,
+            }),
+        }),
+    ),
+
+    /** Maximum amount of visible link previews. */
+    maxAmountOfPreviews: PropTypes.number,
+};
+
+const defaultProps = {
+    linkMetadata: [],
+    maxAmountOfPreviews: 3,
+};
+
+const LinkPreviewer = ({linkMetadata, maxAmountOfPreviews}) => (
```

Comment:
I think maybe something like this is a bit more consistent with the current style guide for how to treat the props in a function component though maybe there is a lot of inconsistency here in the codebase 😅 

```suggestion
function LinkPreviewer(props) {
    const linkMetadata = props.linkMetadata;
    const maxAmountOfPreviews = props.maxAmountOfPreviews;
    return _.map(_.take(uniqBy(linkMetadata, 'url'), maxAmountOfPreviews), (linkData) => {
```